docs/Custom_Device_Types.md

```diff
@@ -0,0 +1,184 @@
+
+1.	Initialize – Same as Inline HW Interface
+2.	Start – Same as Inline HW Interface
+3.	Execute Model
+4.	Close – Same as Inline HW Interface
+
+This custom device is run in-line with the PCL, which calls each case at a specific time with respect to the other components in the system. The PCL will not proceed until the custom device case has returned.
+
+### Execute Model
+
+The execute model case is called in the middle of the PCL. This is the one state of this device that executes during the PCL. This state reads input data, performs a calculation, and then writes output data to VeriStand. Using the Inline Model Interface mode enables you to process data acquired from hardware inputs and send the processed values to hardware outputs with no latency.
+
+![](images/Execute_Model_State.jpg)
+
+**Figure: Execute Model State of the Inline Model Interface Framework**
+
+Do not call **Get** or **Set Channel Value by Data Reference** outside the inline driver VI. Doing so could cause system instability or errors.
+ 
+### Pre-defined Custom Device Types
+
+The type of a custom device refers to its execution mode, which defines how the device interacts with the VeriStand Engine.
+
+The **[following table](https://www.ni.com/documentation/en/veristand/latest/manual/custom-device-types/)** displays the pre-defined custom device types that are included in VeriStand.
+ 
+### Outline of PCL Iteration
+
+The order of operations in the **[Primary Control Loop](https://www.ni.com/documentation/en/veristand/latest/manual/vs-engine/)** varies with respect to the execution mode of the controller. You can adjust this setting in **[System Explorer](https://www.ni.com/documentation/en/veristand/latest/manual/environment/)** » **[Targets](https://www.ni.com/documentation/en/veristand/latest/manual/specify-target/)** » **[Controller](https://www.ni.com/documentation/en/veristand/latest/manual/configure-veristand-engine/)** » Other Settings » Execution Mode.
+
+The **[Data Processing Loop](https://www.ni.com/documentation/en/veristand/latest/manual/vs-engine/)** is responsible for executing procedures, alarms, and calculated channels. For more information about hardware timing in VeriStand see **[KnowledgeBase: Hardware I/O Latency Times in NI VeriStand](https://knowledge.ni.com/KnowledgeArticleDetails?id=kA00Z000000kKVGSA2&l=en-US)**.
+
+### Parallel Mode
+
+In *Parallel* mode, the PCL initiates execution of models and continues to its next iteration without waiting for models to finish executing. This causes a one-cycle delay between when a model executes and when the data it produces is available to the system.
+
+1.	Get inputs from hardware devices in the system definition.
+&nbsp;&nbsp;&nbsp; **Note:** If the system includes an inline hardware interface custom device, the PCL reads DAQ digital lines and counters after the Read Data from HW case of the custom device executes in step 3.
```

Comment:
Is is possible to format these 'Note" statements differently?  It is really hard to read this text list.

Better would be to omit these Notes from the numbered lists, so that it reads:
Get inputs from hardware devices in the system definition.     Note: If the system includes an inline hardware interface custom device, the PCL reads DAQ digital lines and counters after the Read Data from HW case of the custom device executes in step 3.

1. Get inputs from hardware devices in the system definition
2. Read asynchronous custom device FIFOs from the previous iteration*
3. Runs the Read Data From HW case of inline hardware interface custom devices†
4. Reads previous iteration data from models in the system definition‡
5. Reads data from the previous iteration of the Data Processing Loop
6. Processes system mappings§
7. Runs the Execute Model case of inline model interface custom devices
8. Executes steps of running real-time sequences‖
9. Processes system mappings
10. Writes data to models

and then list the Notes as [footnote labels](https://en.wikipedia.org/wiki/Note_(typography)) to refer to the 5 Note statements.  *, †, ‡, §, ‖

If you don't like the footnote approach, you could also turn this list into a 2-column data table where the 2nd column is called 'Notes' or 'Comments' and then you only fill in 5 of the rows that have notes.

Reply:
It is the same format as on the VeriStand Manual from ni.com:
https://www.ni.com/documentation/en/veristand/latest/manual/pcl-execution/